lib/facility.rb

```diff
@@ -1,14 +1,84 @@
+require './lib/vehicle'
+require './lib/registrant'
+require 'date'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :collected_fees, :registered_vehicles
+  attr_accessor :services
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(hash, collected_fees = 0)
+    @name = hash[:name]
+    @address = hash[:address]
+    @phone = hash[:phone]
     @services = []
+    @collected_fees = collected_fees
+    @registered_vehicles = []
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
+    @services = @services.uniq
   end
-end
+
+   def register_vehicle(vehicle)
+    if @services.include?("Vehicle Registration")
+      if vehicle.registration_date.nil?
+        @registered_vehicles << vehicle
+        vehicle.registration_date = Date.today
+        if vehicle.antique?
+          vehicle.plate_type = :antique
+          @collected_fees+=25
+        elsif vehicle.electric_vehicle?
+          vehicle.plate_type = :ev
+          @collected_fees+=200
+        else
+          vehicle.plate_type = :regular
+          @collected_fees+=100
+        end
+        vehicle
+      else
+        "Oops! Looks like this vehicle has already been registered on #{vehicle.registration_date}."
+      end
+     end
+   end
```

Comment:
Your logic here looks really good! Something that could be fun to look into is creating helper methods to slim up this method, and align more with SRP (single responsibility principle). One thing I might do to refactor this method is create 3 more methods specifically for registering different types of vehicles. If you pulled out some of this logic into three other methods, your code could look like: 

```ruby
 @registered_vehicles << vehicle
        vehicle.registration_date = Date.today
        if vehicle.antique?
          register_antique_vehicle(vehicle)
        elsif vehicle.electric_vehicle?
          register_ev_vehicle(vehicle)
        else
          register_reg_vehicle(vehicle)
        end
```